Behaviour change between 7.1 and 7.2 (and later)

<!--
  Issues are so 🔥

  If you remove or skip this template, you'll make the 🐼 sad and the mighty god
  of Github will appear and pile-drive the close button from a great height
  while making animal noises.

  👉🏽 Need support, advice, or help? Don't open an issue!
  Head to StackOverflow or https://gitter.im/webpack/webpack.
-->
As discussed in https://github.com/readthedocs/sphinx_rtd_theme/pull/808 there appears to be a change in the way files are resolved which doesn't seem to correspond to a documented change in sass-loader. It may be a bug in sass-loader or a bug in the way the project that depends on sass-loader is configured. To simplify I have made a minimal project which shows this.
https://github.com/pedro-w/sass-test

- Operating System: Debian 10.1 Buster
- Node Version: v10.16.3
- NPM Version: 6.9.0
- webpack Version: 4.30.0
- sass-loader Version: 7.x (various)

### Expected Behavior

Example project should compile on all 7.x versions

### Actual Behavior

In 7.1 and earlier it is OK, for 7.2 and later an error occurs
```
ERROR in ./src/theme.sass
Module not found: Error: Can't resolve '../fonts/fontawesome-webfont.eot' in '/home/pedro-w/Projects/sass-test/src'
 @ ./src/theme.sass 5:38-81
 @ multi ./src/index.js ./src/theme.sass
```

### Code

https://github.com/pedro-w/sass-test/blob/d8b458430fc04fe7ef24d083027ac5356d4bc00d/webpack.prod.js#L1-L44

### How Do We Reproduce?
Clone the repo above and attempt to build. Change pinned version to 7.1 and the project will now build (note: the output is not useful, it's just for testing)

Made it work with version 8, by putting options under the sassOptions key
```
diff --git a/webpack.prod.js b/webpack.prod.js
index f8b4fae..a21bf74 100644
--- a/webpack.prod.js
+++ b/webpack.prod.js
@@ -17,9 +17,11 @@ module.exports = {
                     {
                         loader: "sass-loader?indentedSyntax",
                         options: {
-                            includePaths: [
-                                "node_modules/font-awesome/scss"
-                            ]
+                            sassOptions: {
+                                includePaths: [
+                                    "node_modules/font-awesome/scss"
+                                ]
+                            }
                         }
                     }
                 ]
```
Now I get
```
ERROR in ./src/theme.sass
Module not found: Error: Can't resolve '../fonts/fontawesome-webfont.eot' in '/private/tmp/sass-test/src'
 @ ./src/theme.sass 5:38-81
 @ multi ./src/index.js ./src/theme.sass
```
as before.

Maybe the webpack config was just wrong before and only worked on <=7.1 by luck?

--- a/src/index.js
+++ b/src/index.js
@@ -40,7 +40,14 @@ function loader(content) {
       : true;
 
   if (shouldUseWebpackImporter) {
+    let modules = ['...'];
+
+    if (options.sassOptions && options.sassOptions.includePaths) {
+      modules = options.sassOptions.includePaths.concat(modules);
+    }
+
     const resolve = this.getResolve({
+      modules,
       mainFields: ['sass', 'style', 'main', '...'],
       mainFiles: ['_index', 'index', '...'],
       extensions: ['.scss', '.sass', '.css', '...'],

--- a/test/helpers/getCodeFromSass.js
+++ b/test/helpers/getCodeFromSass.js
@@ -239,6 +239,15 @@ function getCodeFromSass(testId, options) {
     testFolder,
     'node_modules/another/module.scss'
   );
+  const pathToPackageWithStyleFieldAndCss = isSass
+    ? path.resolve(
+        testFolder,
+        'node_modules/package-with-style-field-and-css/sass/package-with-style-field-and-css.sass'
+      )
+    : path.resolve(
+        testFolder,
+        'node_modules/package-with-style-field-and-css/scss/package-with-style-field-and-css.scss'
+      );
 
   // Pseudo importer for tests
   function testImporter(url) {
@@ -637,6 +646,10 @@ function getCodeFromSass(testId, options) {
       // eslint-disable-next-line no-param-reassign
       url = url
         .replace(/^path-to-alias/, pathToAlias)
+        .replace(
+          /^package-with-style-field-and-css/,
+          pathToPackageWithStyleFieldAndCss
+        )
         .replace(/^~scss-sass-field/, pathToSCSSSassField)
         .replace(/^~sass-sass-field/, pathToSassSassField)
         .replace(/^~scss-style-field/, pathToSCSSStyleField)

--- a/test/loader.test.js
+++ b/test/loader.test.js
@@ -956,6 +956,27 @@ describe('loader', () => {
           expect(getWarnings(stats)).toMatchSnapshot('warnings');
           expect(getErrors(stats)).toMatchSnapshot('errors');
         });
+
+        it(`should watch firstly in the "includePaths" values (${implementationName}) (${syntax})`, async () => {
+          const testId = getTestId('prefer-include-paths', syntax);
+          const options = {
+            sassOptions: {
+              includePaths: [
+                `node_modules/package-with-style-field-and-css/${syntax}`,
+              ],
+            },
+            implementation: getImplementationByName(implementationName),
+          };
+          const compiler = getCompiler(testId, { loader: { options } });
+          const stats = await compile(compiler);
+          const codeFromBundle = getCodeFromBundle(stats, compiler);
+          const codeFromSass = getCodeFromSass(testId, options);
+
+          expect(codeFromBundle.css).toBe(codeFromSass.css);
+          expect(codeFromBundle.css).toMatchSnapshot('css');
+          expect(getWarnings(stats)).toMatchSnapshot('warnings');
+          expect(getErrors(stats)).toMatchSnapshot('errors');
+        });
       }
     });
   });

--- a/test/sass/prefer-include-paths.sass
+++ b/test/sass/prefer-include-paths.sass
@@ -1,0 +1,1 @@
+@import package-with-style-field-and-css

--- a/test/scss/prefer-include-paths.scss
+++ b/test/scss/prefer-include-paths.scss
@@ -1,0 +1,1 @@
+@import "package-with-style-field-and-css";

